src/utils/logging.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export interface LogInterface {
+    debug(msg: string, ...supportingData: any[]): void;
+    warn(msg: string, ...supportingData: any[]): void;
+    error(msg: string, ...supportingData: any[]): void;
+    info(msg: string, ...supportingData: any[]): void;
+}
+
+export enum LogLevel {
+    debug,
+    info,
+    warn,
+    error
+}
+
+
+export class Log implements LogInterface {
+    public currentLogLevel = LogLevel.info;
+    private exitonerror = true;
```

Comment:
Should be `throwOnError` (as it throws on error)

Reply:
Also, no reason to initialize / declare it here - because you already declare it in the constructor

Reply:
Done